tffm/base.py

```diff
@@ -190,119 +191,108 @@ def initialize_session(self):
                 full_log_path = os.path.abspath(self.log_dir)
                 print('Initialize logs, use: \ntensorboard --logdir={}'.format(full_log_path))
         self.session = tf.Session(config=self.session_config, graph=self.core.graph)
-        self.session.run(self.core.init_all_vars)
+        self.session.run([self.core.init_local_vars, self.core.init_global_vars])
 
     @abstractmethod
     def preprocess_target(self, target):
         """Prepare target values to use."""
 
-    def fit(self, X_, y_, s_=None, X_v=None, y_v=None, n_epochs=None, show_progress=False):
-        if self.core.n_features is None:
-            self.core.set_num_features(X_.shape[1])
-
-        assert self.core.n_features==X_.shape[1], 'Different num of features in initialized graph and input'
+    def fit(
+        self, train_directory, valid_directory, n_epochs=None, show_progress=False
+    ):
+        train_tfrecords = ['%s/*.tfrecords.deflate' % train_directory]
+        valid_tfrecords = ['%s/*.tfrecords.deflate' % valid_directory]
 
         if self.core.graph is None:
             self.core.build_graph()
             self.initialize_session()
 
-        if s_ is not None:
-            assert "weighted" in self.core.loss.name, 'Use weighted version of %s by setting parameter use_weights to True' % self.core.loss.name
-
-        used_y = self.preprocess_target(y_)
-
         if n_epochs is None:
             n_epochs = self.n_epochs
 
-        # For reproducible results
-        if self.seed:
-            np.random.seed(self.seed)
-
         if self.verbose > 1:
-            if X_v is not None and y_v is not None:
-                # split validation data into pos/neg examples
-                X_v_neg, y_v_neg, X_v_pos, y_v_pos = self.split_pos_neg(X_v, y_v)
-                self.print_stats(-1, X_v_neg, y_v_neg, X_v_pos, y_v_pos)
             for idx, w in enumerate(self.weights):
                 print('[epoch -1]: weight {}: {:02.3f}'.format(idx, LA.norm(w)))
 
         # Training cycle
         for epoch in tqdm(range(n_epochs), unit='epoch', disable=(not show_progress)):
-            # generate permutation
-            perm = np.random.permutation(X_.shape[0])
-            if s_ is not None:
-                s_ = s_[perm]
+            # initialize iterator and reset accuracy
+            self.session.run(
+                [self.core.iterator.initializer, self.core.init_local_vars],
+                feed_dict={self.core.tfrecord_placeholder: train_tfrecords}
+            )
+
             epoch_loss = []
             weight_norm = [[] for x in range(self.core.order)]
             bias_norm = []
 
             # iterate over batches
-            for bX, bY, bS in batcher(X_[perm], y_=used_y[perm], s_=s_, batch_size=self.batch_size):
```

Comment:
Since `batcher` is removed all of this:

https://github.com/aloisg/tffm/blob/6325da627e323a441d827f6bc00412c2235e2312/tffm/base.py#L17-#L91

can also be deleted.

Reply:
Actually, it's still used for prediction. Do we want to keep `def predict(self, X, pred_batch_size=None)` in `models.py` ?

Reply:
I agree, let's keep it!